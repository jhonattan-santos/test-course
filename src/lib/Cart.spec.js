import { Cart } from './Cart';

describe('Cart', () => {
	let cart;
	const product = {
		name: 'Adidas shoes',
		price: 39989,
	};

	const product2 = {
		name: 'Adidas shoes for woman',
		price: 29989,
	};

	beforeEach(() => {
		cart = new Cart();
	});

	it('should return 0 when getTotal() is eecuted in a newly created instance', () => {
		expect(cart.getTotal()).toEqual(0);
	});

	it('should multiply quntity and price and receive the total amounth', () => {
		const item = {
			product,
			quantity: 2,
		};

		cart.add(item);
		expect(cart.getTotal()).toEqual(79978);
	});

	it('should ensure no more than one product exists at a time', () => {
		cart.add({
			product,
			quantity: 2,
		});

		cart.add({
			product,
			quantity: 1,
		});

		expect(cart.getTotal()).toEqual(39989);
	});

	it('should update total when a product gets included and then remove one', () => {
		cart.add({
			product,
			quantity: 2,
		});
		cart.add({
			product: product2,
			quantity: 1,
		});

		expect(cart.getTotal()).toEqual(109967);

		cart.remove(product);

		expect(cart.getTotal()).toEqual(29989);
	});
});
