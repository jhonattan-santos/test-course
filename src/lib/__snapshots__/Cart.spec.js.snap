// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout() should return an object with the total and a list of items 1`] = `
Object {
  "products": Array [
    Object {
      "product": Object {
        "name": "Adidas shoes",
        "price": 39989,
      },
      "quantity": 2,
    },
    Object {
      "product": Object {
        "name": "Adidas shoes for woman",
        "price": 29989,
      },
      "quantity": 1,
    },
  ],
  "total": 109967,
}
`;

exports[`Cart should return an object with the total when summary() is called 1`] = `
Object {
  "products": Array [
    Object {
      "product": Object {
        "name": "Adidas shoes",
        "price": 39989,
      },
      "quantity": 2,
    },
    Object {
      "product": Object {
        "name": "Adidas shoes for woman",
        "price": 29989,
      },
      "quantity": 1,
    },
  ],
  "total": 109967,
}
`;
